python/lsst/drp/tasks/assemble_cell_coadd.py

```diff
@@ -80,8 +89,18 @@ class AssembleCellCoaddConnections(
         dimensions=("tract", "patch", "band", "skymap"),
     )
 
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config is None:
+            return
+
+        if not config.do_apply_artifact_mask:
+            self.inputs.remove("artifactMasks")
+
 
 class AssembleCellCoaddConfig(PipelineTaskConfig, pipelineConnections=AssembleCellCoaddConnections):
+    do_apply_artifact_mask = Field[bool](doc="Apply artifact masks to the input warps?", default=True)
```

Comment:
I introduced this as a way to remove `artifactMask` as an input to making cell-based coadds on simulated data. However, this had a surprising (but correct) behavior - if excluded some quanta that didn't have PSF-matched warps and hence artifactMask. This is potentially very confusing to those who don't understand the middleware. I was going to work some solution out in `adjustQuantum` by selecting only those quanta with PSF-matched warps. But for the simulations, those are not bound to be present either. So I am inclined to remove this option together and expect a trivial all-zero `artifactMask` instead.

Reply:
I don't find it confusing, but if it doesn't solve the problem you wanted it to solve (or any other problem) then I agree it's dead weight.